File: euler/001/001.c
#include <stdio.h>
/* jsu
created 16 Dec 2012
last edit 16 Dec 2012
***correct solution***
*/

int main(){
  int x = 3;
  int y = 5;
  int limit = 1000;
  int i;
  int total = 0;
  for (i = 1; i < limit; i++){
	if (i % x == 0) {
	  	total += i;
		printf("%d\n", i);}
	else if (i % y == 0){
	  	total += i;
		printf("%d\n", i);
	  }
  }
  printf ("%d\n", total);
}

File: euler/001/001.py
'''
jsu
created 16 Dec 2012
last edited 16 Dec 2012
*** correct solution***
'''

def sum_multiples(multiplier, limit):
	'''sum the multiples of multiplier under limit'''      
	values = []
	x = 1
	while (x * multiplier) < limit:
#		print "added 	
		values.append(x * multiplier)
		x += 1

	return values

if __name__ == '__main__':
	'''run as main'''
	LIMIT = 1000
	three = sum_multiples(3, LIMIT) 
	five = sum_multiples(5, LIMIT)
	
	for f in five:
		if f not in three: 
			three.append(f)

	sum = 0
	for val in three:
		sum += val

	print sum
	

